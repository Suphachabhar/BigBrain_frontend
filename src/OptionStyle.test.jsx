import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import { OptionStyle } from './components/OptionStyle.styles';

describe('OptionStyle', () => {
  const noop = () => {};

  // test the optionStyle component
  it('trigger onChange event handler when clicked', () => {
    const onChange = jest.fn();
    shallow(<OptionStyle onChange="{onChange}" />).simulate('change');
    expect(onChange).toHaveBeenCalledTimes(0);
  });

  it('renders with minimal props', () => {
    const selectFunc = renderer.create(<OptionStyle onChange={noop} />).toJSON;
    expect(selectFunc).toMatchSnapshot();
  });
});
