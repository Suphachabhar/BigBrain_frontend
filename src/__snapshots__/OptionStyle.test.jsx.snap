// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionStyle renders with minimal props 1`] = `[Function]`;
